src/DataContainers.jl

```diff
@@ -19,11 +19,10 @@ struct DataContainer{FT <: Real}
     stored_data::AbstractMatrix{FT}
     #constructor with 2D arrays
     function DataContainer(stored_data::AbstractMatrix{FT}; data_are_columns = true) where {FT <: Real}
-
         if data_are_columns
-            new{FT}(stored_data)
+            new{FT}(deepcopy(stored_data))
         else
-            new{FT}(permutedims(stored_data, (2, 1)))
+            new{FT}(permutedims(deepcopy(stored_data), (2, 1)))
```

Comment:
This one is not necessary, permutedims already performs a deep copy before transposing.

Reply:
I thought it would be clearer this way, I'll add a comment instead